src/docs/development/platform-integration/add-to-app-android/add-flutter-fragment.md

```diff
@@ -0,0 +1,311 @@
+---
+title: Adding a Flutter Fragment to an Android app
+short-title: Add a Flutter Fragment
+description: Learn how to add a Flutter Fragment to your existing Android app.
+---
+
+{% asset
+development/platform-integration/add-to-app-android/add-flutter-fragment/add-flutter-fragment_header.png
+class="mw-100" alt="Add Flutter Fragment Header" %}
+
+This guide describes how to add a Flutter `Fragment` to an existing Android app.
+In Android, a [`Fragment`] represents a modular piece of a larger UI. A
+`Fragment` might be used to present a sliding drawer, tabbed content, a page in
+a `ViewPager`, or it might simply represent a normal screen in a
+single-`Activity` app. Flutter provides a `FlutterFragment` so that developers
+can present a Flutter experience any place that they can use a regular
+`Fragment`.
+
+[`Fragment`]: https://developer.android.com/guide/components/fragments
+
+If an `Activity` is equally applicable for your application needs, consider
+[using a `FlutterActivity`] instead of a `FlutterFragment`, which is quicker and
+easier to use.
+
+[using a `FlutterActivity`]: /docs/development/platform-integration/add-to-app-android/add-flutter-screen
+
+`FlutterFragment` allows developers to control the following details of the
+Flutter experience within the `Fragment`:
+
+ * Initial Flutter route
+ * Dart entrypoint to execute
+ * Opaque vs translucent background
+ * Whether `FlutterFragment` should control its surrounding `Activity`
+
+`FlutterFragment` also comes with a number of calls that must be forwarded from
+its surrounding `Activity`. These calls allow Flutter to react appropriately to
+OS events.
+
+All varieties of `FlutterFragment`, and its requirements, are described in this
+guide.
+
+## Add a `FlutterFragment` to an `Activity`
+
+The first thing to do to use a `FlutterFragment` is to add it to a host
+`Activity`.
+
+To add a `FlutterFragment` to a host `Activity`, start by instantiating and
+attaching an instance of `FlutterFragment` in `onCreate()` within the
+`Activity`, or at some other time that works for your app:
+
+```java
```

Comment:
@sfshaza2 do we have a nice way of presenting a code box but with a toggle so users can choose to see the snippet in either Java or Kotlin?

Reply:
Not the code box, per se. But we definitely support tabs and each tab could have it’s own code box. Yo can see what I mean here:  https://flutter.dev/docs/get-started/editor?tab=androidstudio
And the src is in /docs/get-started/editor.md

This would be a good feature request for the site, but we don’t have an infrastructure person to implement it at the moment.